Streetcode/Streetcode.BLL/MediatR/Partners/GetToUpdateByStreetcodeId/GetPartnersToUpdateByStreetcodeIdHandler.cs

```diff
@@ -0,0 +1,45 @@
+using AutoMapper;
+using FluentResults;
+using MediatR;
+using Streetcode.BLL.DTO.Partners;
+using Microsoft.EntityFrameworkCore;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+
+namespace Streetcode.BLL.MediatR.Partners.GetByStreetcodeIdToUpdate
+{
+    public class GetPartnersToUpdateByStreetcodeIdHandler : IRequestHandler<GetPartnersToUpdateByStreetcodeIdQuery, Result<IEnumerable<PartnerDTO>>>
+    {
+        private readonly IMapper _mapper;
+        private readonly IRepositoryWrapper _repositoryWrapper;
+
+        public GetPartnersToUpdateByStreetcodeIdHandler(IMapper mapper, IRepositoryWrapper repositoryWrapper)
+        {
+            _mapper = mapper;
+            _repositoryWrapper = repositoryWrapper;
+        }
+
+        public async Task<Result<IEnumerable<PartnerDTO>>> Handle(GetPartnersToUpdateByStreetcodeIdQuery request, CancellationToken cancellationToken)
+        {
+            var streetcode = await _repositoryWrapper.StreetcodeRepository
+                .GetSingleOrDefaultAsync(st => st.Id == request.StreetcodeId);
+
+            if (streetcode is null)
+            {
+                return Result.Fail(new Error($"Cannot find any streetcode with corresponding streetcode id: {request.StreetcodeId}"));
+            }
+
+            var partners = await _repositoryWrapper.PartnersRepository
+                .GetAllAsync(
+                    predicate: p => p.Streetcodes.Any(sc => sc.Id == streetcode.Id),
+                    include: p => p.Include(pl => pl.PartnerSourceLinks));
+
+            if (partners is null)
+            {
+                return Result.Fail(new Error($"Cannot find a coordinates by a streetcode id: {request.StreetcodeId}"));
+            }
```

Comment:
set the correct error message. And also for what do you need this request?

Reply:
done. I need this request because the handler GetPartnersByStreetcodeId returns partners that are visible everywhere. However, I don't need them for update